pycqed/analysis_v2/timedomain_analysis.py

```diff
@@ -289,11 +289,66 @@ def extract_data(self):
         if len(self.channel_map) == 0:
             raise ValueError('No qubit RO channels have been found.')
 
-        self.data_to_fit = deepcopy(self.get_param_value('data_to_fit', {}))
+        # whether to rotate data or not
+        self.rotate = self.get_param_value('rotate', default_value=True)
+
+        # to be used if data_type == singleshot (see process_data)
+        self.predict_proba = self.get_param_value("predict_proba", False)
+        if self.predict_proba and self.get_param_value("classified_ro", False):
+            log.warning("predict_proba set to 'False' as probabilities are"
+                        "already obtained from classified readout")
+            self.predict_proba = False
+        # ensure rotation is removed when single shots yield probabilities
+        if self.get_param_value("classified_ro", False) or self.predict_proba:
+            self.rotate = False
+
+        if self.get_param_value('global_PCA') is not None:
+            log.warning('Parameter "global_PCA" is deprecated. Please set '
+                        'rotation_type="global_PCA" instead.')
+        self.rotation_type = self.get_param_value(
+            'rotation_type',
+            default_value='cal_states' if self.rotate else 'no_rotation')
 
         # creates self.sp
         self.get_sweep_points()
 
+        # creates self.cp if found else defaults to legacy extraction
+        self.get_cal_points()
+
+        # get data_to_fit
+        self.data_to_fit = deepcopy(self.get_param_value('data_to_fit'))
+        if self.data_to_fit is None:
+            # If we have cal points, but data_to_fit is not specified,
+            # choose a reasonable default value. In cases with only two cal
+            # points, this decides which projected plot is generated. (In
+            # cases with three cal points, we will anyways get all three
+            # projected plots.)
+            if 'e' in self.cal_states_dict.keys():
+                self.data_to_fit = {qbn: 'pe' for qbn in self.qb_names}
+            elif 'g' in self.cal_states_dict.keys():
+                self.data_to_fit = {qbn: 'pg' for qbn in self.qb_names}
+            else:
+                self.data_to_fit = {}
+        # TODO: Steph 15.09.2020
```

Comment:
Are there plans to resolve this TODO?

Reply:
No :D This is only used by the cphase analysis (I needed to hack this when Christian wanted to get the amount of swap errors and he needed to fit multiple projected data for the same qubit). It would break all the analyses we have and it will be a lot of work to upgrade them. Maybe at some point when we have a further use case.